src/test/java/com/squareup/phrase/ListPhraseTest.java

```
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.phrase;
+
+import com.squareup.phrase.ListPhrase.Formatter;
+import edu.emory.mathcs.backport.java.util.Collections;
+import java.util.Arrays;
+import java.util.HashSet;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+@RunWith(RobolectricTestRunner.class) @Config(manifest = Config.NONE)
+public class ListPhraseTest {
+  private final static String twoElementSeparator = " and ";
+  private final static String nonFinalElementSeparator = ", ";
+  private final static String finalElementSeparator = ", and ";
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  ListPhrase phrase;
+
+  @Before public void setUp() {
+    phrase = ListPhrase.from(twoElementSeparator, nonFinalElementSeparator, finalElementSeparator);
+  }
+
+  @Test public void joinEmptyListReturnsEmptyString() {
```

Again, wonder if this is doesn't amount to a silent failure.